perun-core/src/main/java/cz/metacentrum/perun/core/entry/ResourcesManagerEntry.java

```diff
@@ -99,6 +99,32 @@ public Resource createResource(PerunSession sess, Resource resource, Vo vo, Faci
 		return getResourcesManagerBl().createResource(sess, resource, vo, facility);
 	}
 
+	public Resource copyResource(PerunSession sess, Resource templateResource, Resource destinationResource, boolean withGroups) throws InternalErrorException, ResourceNotExistsException, FacilityNotExistsException, PrivilegeException, VoNotExistsException, ResourceExistsException {
+		Utils.checkPerunSession(sess);
+		Vo destinationVo = getPerunBl().getVosManagerBl().getVoById(sess, destinationResource.getVoId());
+		Facility destinationFacility = getPerunBl().getFacilitiesManagerBl().getFacilityById(sess, destinationResource.getFacilityId());
+		getPerunBl().getVosManagerBl().checkVoExists(sess, destinationVo);
+		getPerunBl().getFacilitiesManagerBl().checkFacilityExists(sess, destinationFacility);
```

Comment:
This is not needed. The Vo was selected from DB by it's ID so it should exist. Same for the Facility.